packages/components/src/alignment-matrix-control/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import classnames from 'classnames';
+import {
+	unstable_useCompositeState as useCompositeState,
+	unstable_Composite as Composite,
+	unstable_CompositeGroup as CompositeGroup,
+} from 'reakit';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useInstanceId } from '@wordpress/compose';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Cell from './cell';
+import { Root, Row } from './styles/alignment-matrix-control-styles';
+import { useRTL } from '../utils/rtl';
+import AlignmentMatrixControlIcon from './icon';
+import { GRID, getItemId } from './utils';
+
+function useBaseId( id ) {
+	const instanceId = useInstanceId( AlignmentMatrixControl );
+	const prefix = id || 'alignment-matrix-control';
+
+	return `${ prefix }-${ instanceId }`;
+}
+
+export default function AlignmentMatrixControl( {
+	className,
+	id,
+	label = __( 'Alignment Matrix Control' ),
+	defaultValue = 'center center',
+	value,
+	onChange = noop,
+	width = 92,
+	...props
+} ) {
+	const [ immutableDefaultValue ] = useState( value ?? defaultValue );
+	const isRTL = useRTL();
+	const baseId = useBaseId( id );
+	const initialCurrentId = getItemId( baseId, immutableDefaultValue );
+
+	const composite = useCompositeState( {
+		baseId,
+		currentId: initialCurrentId,
+		rtl: isRTL,
+	} );
+
+	const handleOnChange = ( nextValue ) => {
+		onChange( nextValue );
+	};
+
+	useEffect( () => {
+		if ( typeof value !== 'undefined' ) {
+			composite.setCurrentId( getItemId( baseId, value ) );
+		}
+	}, [ value, composite.setCurrentId ] );
+
+	const classes = classnames(
+		'component-alignment-matrix-control',
+		className
+	);
+
+	return (
+		<Composite
+			{ ...props }
+			{ ...composite }
+			aria-label={ label }
+			as={ Root }
+			className={ classes }
+			role="grid"
+			width={ width }
+		>
+			{ GRID.map( ( cells, index ) => (
+				<CompositeGroup
+					{ ...composite }
+					as={ Row }
+					role="row"
+					key={ index }
+				>
+					{ cells.map( ( cell ) => {
+						const cellId = getItemId( baseId, cell );
+						const isActive = composite.currentId === cellId;
+
+						return (
+							<Cell
+								{ ...composite }
+								id={ cellId }
+								isActive={ isActive }
+								key={ cell }
+								value={ cell }
+								onFocus={ () => handleOnChange( cell ) }
+								onClick={ () =>
+									// VoiceOver doesn't focus elements on click
+									composite.move( cellId )
+								}
+							/>
+						);
+					} ) }
+				</CompositeGroup>
+			) ) }
+		</Composite>
+	);
+}
+
+AlignmentMatrixControl.Icon = AlignmentMatrixControlIcon;
+AlignmentMatrixControl.icon = <AlignmentMatrixControlIcon />;
```

Comment:
Very weird to expose both the component and the element? do we really need the element?

Reply:
I just added this comment:

```
/**
 * Exported rendered icon, which can be used with the <Icon /> component
 * from @wordpress/icons.
 *
 * Example:
 * <Icon icon={ AlignmentMatrixControl.icon } />
 */
```

The idea was to respect how the `<Icon />` component's `icon` prop worked, which expects elements for it to render correctly.

Let me know if this makes sense! I'm open and happy to make changes

Reply:
I still find it confusing personally to export the same thing twice. One way to solve this is

```
<Icon icon={ <AlignmentMatrixControl.Icon /> } />
```

Reply:
We totally could! Does it feel strange that the following are inconsistent? (I'm not opposed, just wanted to double check)

```
<Icon icon={ <AlignmentMatrixControl.Icon /> } />

<Icon icon={ chevronLeft } />
```

Reply:
I personally prefer it this way, `<Icon icon={ <AlignmentMatrixControl.Icon /> } />` This is totally reasonable for me. In fact we already do that for somethings. I believe we had something like that for the heading icons at some point (levels)

Reply:
Cool! Works for me :D. I'll make that update